strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -464,28 +264,25 @@ public void deleteMetadata(String storageId,
 
         Storage storage = getConfiguration().getStorage(storageId);
         Repository repository = storage.getRepository(repositoryId);
-        StorageProvider storageProvider = storageProviderRegistry.getProvider(repository.getImplementation());
-
-        final File repoPath = storageProvider.getFileImplementation(repository.getBasedir());
-
+        RepositoryPath repositoryPath = resolve(repository);
+        if (!Files.isDirectory(repositoryPath))
```

Comment:
I'm afraid I don't quite agree with this change. What you're doing in this case is changing the logic/behaviour, as far as I can see. My code is specifically for directories, whereas this is ignoring directories, as far as I understand.

 Could you please do a diff with my original code and make sure this is right, or explain what this is correcting, if I'm misunderstanding you...? Thanks! :)